printing/printer.ml

```diff
@@ -211,16 +211,20 @@ let pr_universe_ctx sigma ?variance c =
   else
     mt()
 
-let pr_abstract_universe_ctx sigma ?variance c =
-  if !Detyping.print_universes && not (Univ.AUContext.is_empty c) then
+let pr_abstract_universe_ctx sigma ?variance c local =
+  let open Univ in
+  if !Detyping.print_universes && not (Univ.AUContext.is_empty c && Univ.ContextSet.is_empty local) then
+    let local = if ContextSet.is_empty local then mt() else
+        str "local: " ++ Univ.pr_universe_context_set (Termops.pr_evd_level sigma) local
+    in
     fnl()++pr_in_comment (fun c -> v 0
-      (Univ.pr_abstract_universe_context (Termops.pr_evd_level sigma) ?variance c)) c
+      (Univ.pr_abstract_universe_context (Termops.pr_evd_level sigma) ?variance c) ++ v 0 local) c
```

Comment:
Could someone who understands printing boxes have a look at this so that we don't get weird indentation (see OP)?

Reply:
The `0` will not give indentation, so that should be ok. However, the printing of universes could be improved. For instance, with `Set Printing Depth 10`, I get things like:
```
a
     : Type@{max(Top.1+1,Top.2+1,Top.3+1,Top.4+1,Top.5+1,Top.6+1,Top.7+1,Top.8+1,Top.9)}
(* Top.1 Top.2 Top.3 Top.4 Top.5 Top.6 Top.7 Top.8 Top.9 Top.10 |= 
   Top.10 < Top.9
    *)
```
To allow breaks, `UContext.pr` could use `hov` (or `hv`) rather than `h` (I also wonder whether the `h 0` in front of `v 0` is really useful). In `Ppconstr.pr_constr`, printing the `max` could use `str"," ++ cut()` to allow breaking.

Otherwise, back to this PR, the two steps printing introducing two `|=` is a bit difficult to understand at first. Alternative proposals I could think of includes:
```
Top.1 {Top.3 Top.2} |= Prop < Set
                       Set < Top.1
                       Top.2 < Top.1
                       Top.3 < Top.1
```
or
```
Top.1 {Top.3 Top.2} |= Prop < Set
                       Set < Top.1
                       {Top.2 < Top.1}
                       {Top.3 < Top.1}
```
or
```
Top.1 Top.Internal.3 Top.Internal.2 |= Prop < Set
                                       Set < Top.1
                                       Top.Internal.2 < Top.1
                                       Top.Internal.3 < Top.1
```
or (`Local` instead of `Internal`), or
```
Top.1 Top.3 (internal) Top.2 (internal) |= Prop < Set
                                           Set < Top.1
                                           Top.2 < Top.1 (internal)
                                           Top.3 < Top.1 (internal)
```
or
```
Public universes:
Top.1 |= Prop < Set
         Set < Top.1
Private universes:
Top.3 Top.2 |= Top.2 < Top.1
               Top.3 < Top.1
```